src/libraries/Common/tests/System/Net/Http/Http3LoopbackConnection.cs

```diff
@@ -44,21 +50,26 @@ public Http3LoopbackConnection(QuicConnection connection)
 
         public override void Dispose()
         {
+            // Close any remaining request streams (but NOT control streams, as these should not be closed while the connection is open)
             foreach (Http3LoopbackStream stream in _openStreams.Values)
             {
                 stream.Dispose();
             }
 
-            if (!_closed)
-            {
-            //    CloseAsync(H3_INTERNAL_ERROR).GetAwaiter().GetResult();
-            }
+            // Dispose the connection
+            // If we already waited for graceful shutdown from the client, then the connection is already closed and this will simply release the handle.
+            // If not, then this will silently abort the connection.
+            _connection.Dispose();
 
-            //_connection.Dispose();
+            // Dispose control streams so that we release their handles too.
+            _inboundControlStream?.Dispose();
```

Comment:
As long as we close the connection first, it shouldn't matter.

That said, in this path we may or may not have closed the connection (as per comment above). If we haven't, then it does matter because if we dispose the stream before the connection, this will abort the stream because the connection is still open. We don't want that.